### PR TITLE
Use composite action to reuse setup steps

### PR DESCRIPTION
**Pull request by ysndr**
All our actions use the same setup.
To make updating easier I implemented a composite action that can be reused by all other actions

(also includes adaptions for the check implemented in #432)

### CODE REVIEW

**Comment by ncfavier** [comment]
Good idea, could we also factor out the checkout step?

Also, why are we using `fetch-depth: 0`? I previously thought this made a shallow clone, but it in fact makes a deep clone, shallow is the default.

**Comment by garbas** [comment]
> Good idea, could we also factor out the checkout step?
> 
> Also, why are we using `fetch-depth: 0`? I previously thought this made a shallow clone, but it in fact makes a deep clone, shallow is the default.

I think this is needed for flakes.

**Review by garbas — APPROVED** [review]

**Comment by ncfavier** [comment]
> > Good idea, could we also factor out the checkout step?
> > Also, why are we using `fetch-depth: 0`? I previously thought this made a shallow clone, but it in fact makes a deep clone, shallow is the default.
> 
> I think this is needed for flakes.

I don't think so, I have similar flakes-based workflows on my own repos and shallow clones are fine. Will make a follow-up PR

File: .github/actions/common-setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Setup Environment
+inputs:
+  CACHIX_SIGNING_KEY: # determines what node version to install
```
Comment (ncfavier) [inline .github/actions/common-setup/action.yml:3]: I don't understand this comment. Why not use `secrets.CACHIX_SIGNING_KEY` directly?

Comment (ysndr) [reply .github/actions/common-setup/action.yml:3]: You can't access secrets from the action.
Also the checkout cant be factored out into the action as the action file has to be present before gh can call it.

**Pull request merged**